WIP: Snaps install UI

ritave opened the pull request:
<img width="478" alt="Screenshot 2021-12-06 at 18 37 09" src="https://user-images.githubusercontent.com/1614945/144894372-76f5ece2-d71a-4da8-8611-5f84488f2205.png">

rekmarks committed:
Add Snaps

* Implement bls-signer support
* Implement filsnap support
* Add hooknames and hook selectors to RPC methods
* Fix filsnap installation on vault creation
* Various build and CI config updates
* Update sourcemap validation
* Update Sentry configuration
* Change name of production environment to 'filecoin-production'
* Ignore useless sentry error in e2e tests
* Update README.md
* Snaps update 0.0.7 (#11854)
* Add iframe execution environment (#11865)
* Remove WebWorkerExecutionEnvironmentService import
* Ignore filsnap bundle in depcheckrc (#11866)
  * This stops the `depcheck` check from failing.
* Restore Firefox compatibility (#11867)
* Notify user of snap initial execution errors (#11901)
* Add alert on snap initial execution errors
* Changed window.alert to be a browser notification
* Add guard for pluginController removePlugin
* Remove Pre-Installed Filsnap (#12111)
* Add more global snap hooks (#12113)
* Changed snap imports to use `@metamask/` org (#12238)
* Remove all uses of @mm-snaps
* snaps packages @0.3.1 (#12610) (squashed)
* Fence all Flask code
* Remove snap_manageAssets and AssetController
* Add locale messages for expected production permissions
* Remove permissionsDescriptions state (replace by locale messages)
* Remove unnecessary modifications to various files
* Update Flask LavaMoat policy
* Fix permission description display
* Restore LavaMoat beta policy
* Fix permission locale message handling
* Fix build errors, regenerate build system policy
* Fix some broken code fences and translation calls
* Added snaps error notification (#12860)
* skunkworks@0.5.0, migrate snap restricted methods
* Clean up untrusted connection subject type logic
* Record Snap subject metadata
* Fix useOriginMetadata React hook
* Backported bug fix from snaps-0.1 to snaps (#13094)
* Added Snaps Custom Confirmation (#13036)
* Fix bug when adding subject metadata (#13102)

Co-authored-by: Erik Marks <rekmarks@protonmail.com>
Co-authored-by: George Marshall <george.marshall@consensys.net>
Co-authored-by: Hassan Malik <hassan.malik@consensys.net>
Co-authored-by: Olaf Tomalka <olaf.tomalka@gmail.com>
Co-authored-by: Shane Jonas <jonas.shane@gmail.com>

ritave committed:
Added Snap Install page to the UI

segun commented on app/_locales/en/messages.json:
```diff
@@ -2483,6 +2495,15 @@
+    "message": "To confirm you understand, check all."
+  },
+  "snapInstallWarningKeyAccess": {
+    "message": "You are granting key access to $1 snap. This is irrevocable and makes $1 able to control your account and assets. Make sure you trust $1 developers."
```
This needs a description of the variables

rekmarks committed:
Snaps skunkworks@0.6.0 (#13076)

Updates all `snaps-skunkworks` packages to `0.6.0`, and uses the `0.3.0` iframe execution environment.

rekmarks committed:
skunkworks@0.6.1

ritave commented:
Will add PR to `snaps` branch instead of `develop`
